frontend/src/metabase/admin/settings/components/SettingsLdapForm.tsx

```diff
@@ -31,118 +37,53 @@ const testParentheses: TestConfig<string | null | undefined> = {
-  key: string; // ensuring key is required
-  is_env_setting?: boolean;
-  env_name?: string;
-  default?: any;
-};
-
-type Props = {
-  elements: LdapFormSettingElement[];
-  settingValues: SettingValues;
-  onSubmit: (values: SettingValues) => void;
-};
-
-export const SettingsLdapFormView = ({
-  elements = [],
-  settingValues,
-  onSubmit,
-}: Props) => {
-  const isEnabled = settingValues["ldap-enabled"];
-
-  const settings = useMemo(() => {
-    return _.indexBy(elements, "key");
-  }, [elements]);
-
-  const fields = useMemo(() => {
-    return _.mapObject(settings, (setting) => ({
-      name: setting.key,
-      label: setting.display_name,
-      description: setting.description,
-      placeholder: setting.is_env_setting
-        ? t`Using ${setting.env_name}`
-        : setting.placeholder || setting.default,
-      default: setting.default,
-      required: setting.required,
-      autoFocus: setting.autoFocus,
-    }));
-  }, [settings]);
-
-  const defaultableAttrs = useMemo(() => {
-    return new Set(PLUGIN_LDAP_FORM_FIELDS.defaultableFormFieldAttributes);
-  }, []);
-
-  const ldapAttributes = useMemo(
-    () => [
-      ...PLUGIN_LDAP_FORM_FIELDS.formFieldAttributes,
-
-      // Server Settings
-      "ldap-host",
-      "ldap-port",
-      "ldap-security",
-      "ldap-bind-dn",
-      "ldap-password",
-
-      // User Schema
-      "ldap-user-base",
-      "ldap-user-filter",
-
-      // Attributes
-      "ldap-attribute-email",
-      "ldap-attribute-firstname",
-      "ldap-attribute-lastname",
-
-      // Group Schema
-      "ldap-group-sync",
-      "ldap-group-base",
-      "ldap-group-membership-filter",
-      "ldap-sync-admin-group",
-    ],
-    [],
-  ) as SettingKey[];
-
-  const attributeValues = useMemo(() => {
-    return getAttributeValues(
-      ldapAttributes,
-      settings,
-      settingValues,
-      defaultableAttrs,
-    );
-  }, [settings, settingValues, ldapAttributes, defaultableAttrs]);
+export type LdapSettings = Pick<
+  EnterpriseSettings,
+  | "ldap-enabled"
+  | "ldap-host"
+  | "ldap-port"
+  | "ldap-security"
+  | "ldap-bind-dn"
+  | "ldap-password"
+  | "ldap-user-base"
+  | "ldap-user-filter"
+  | "ldap-attribute-email"
+  | "ldap-attribute-firstname"
+  | "ldap-attribute-lastname"
+  | "ldap-group-base"
+  | "ldap-group-membership-filter"
+>;
+
+export const SettingsLdapForm = () => {
+  const { data: settingDetails } = useGetAdminSettingsDetailsQuery();
+  const { data: settingValues } = useGetSettingsQuery();
+  const [updateLdapSettings] = useUpdateLdapMutation();
+  const isEnabled = settingValues?.["ldap-enabled"];
 
   const handleSubmit = useCallback(
-    (values: LdapFormValues) => {
-      return onSubmit({
+    (values: Partial<LdapSettings>) => {
+      return updateLdapSettings({
         ...values,
-        "ldap-port": parseInt(values["ldap-port"]?.trim() || ""),
+        "ldap-port": Number(values["ldap-port"]),
```

Comment:
this is now a number input, so we don't need to trim, because you can't type whitespace